pkg/ebpf/tracee.go

```diff
@@ -819,17 +822,12 @@ func (t *Tracee) populateBPFMaps() error {
 	}
 
 	// Initialize tail call dependencies
-	tailCalls := make(map[events.TailCall]bool)
-	for e := range t.events {
-		for _, tailCall := range events.Definitions.Get(e).Dependencies.TailCalls {
-			if tailCall.MapIdx > 10000 { // undefined syscalls (check arm64.go)
-				continue
-			}
-			tailCalls[tailCall] = true
-		}
+	tailCalls, err := t.GetTailCalls()
+	if err != nil {
+		return err
 	}
-	for tailCall := range tailCalls {
-		err := t.initTailCall(tailCall.MapName, tailCall.MapIdx, tailCall.ProgName)
+	for _, tailCall := range tailCalls {
+		err := t.initTailCall(tailCall.MapName, tailCall.MapIndexes, tailCall.ProgName)
```

Comment:
For near future: feels to me that, just like we did with probes, creating a package/interface for it, we could also have a small package for "tailcalls" and remove "initTailCall" from tracee object.

This would avoid having to have []events.TailCalls constructed and passed as an argument to a tracee object method.

Reply:
I actually think tail calls can even be probes themselves since they technically satisfy the interface of Attach and Detach.
There's certainly room for thought here.